salt/utils/url.py

```diff
@@ -47,7 +47,7 @@ def create(path, saltenv=None):
 
     query = f"saltenv={saltenv}" if saltenv else ""
     url = salt.utils.data.decode(urlunparse(("file", "", path, "", query, "")))
-    return "salt://{}".format(url[len("file:///") :])
+    return f'salt://{re.sub(r"file:/*", "", url)}'
```

Comment:
The regex needs a begin anchor `^` so it doesn't replace `file:/` somewhere in the middle of the url: `f'salt://{re.sub(r"^file:/*", "", url)}'